middle_end/base_types/closure_origin.ml

```diff
@@ -0,0 +1,22 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Pierre Chambart, OCamlPro                        *)
+(*     Mark Shinwell, Leo White and Fu Yong Quah, Jane Street Europe      *)
+(*                                                                        *)
+(*   Copyright 2013--2017 OCamlPro SAS                                    *)
+(*   Copyright 2014--2017 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-4-9-30-40-41-42"]
+
+include Closure_id
+
+let create t = t
+let rename f t = f t
```

Comment:
I didn't see any use of rename. I think it is required for building packs.
I'm not sure there is a good way to validate this with Flambda_invariants, but that would be the right way to find such kind of problems here.

Reply:
We (myself, @xclerc and @lpw25) thought about this packing business and came to the conclusion that the existing code for renaming e.g. set of closures IDs isn't actually required -- since the "for-pack" option, required to build units that will go in a pack, will already cause symbols to be generated with the packed module prefix.  As such it seems that no renaming should be necessary for closure origins.  We can deal with removing the existing renaming on the other identifiers in due course.